src/main/java/games/strategy/engine/lobby/server/db/UserController.java

```diff
@@ -65,12 +71,19 @@ public void updateUser(final DBUser user, final HashedPassword hashedPassword) {
     Preconditions.checkArgument(user.isValid(), user.getValidationErrorMessage());
 
     try (final Connection con = connectionSupplier.get();
-        final PreparedStatement ps =
-            con.prepareStatement("update ta_users set password = ?,  email = ? where username = ?")) {
+        final PreparedStatement ps = con.prepareStatement(
+            String.format("update ta_users set %s=?, email=? where username=?", getPasswordColumn(hashedPassword)))) {
       ps.setString(1, hashedPassword.value);
       ps.setString(2, user.getEmail());
       ps.setString(3, user.getName());
       ps.execute();
+      if (!hashedPassword.isBcrypted()) {
```

Comment:
Can this be re-written using the technique you used below in `createUser()`?  That is, where you set both columns at once?  I find the way you wrote the code in `createUser()` to be more readable.  I think that will get rid of the need for the `getPasswordColumn()` method, too.

**EDIT:** I'm assuming you did it this way because, when a user updates their password from a new client, you want both columns to be updated.  Since a new client will make two calls to this method (one with the legacy password and one with the new password), you don't want to set the legacy password to `NULL` during the second call when the new password is set.

However, this logic seems to be dependent on the order in which the new client makes those calls.  That took me a bit to figure out.  It might be good to document in `LobbyMenu` that the order of the `updateUser()` calls is important and can't be reversed.

I wonder if it's possible to send both the MD5Crypted and bcrypted passwords from a new client in a single `updateUser()` call.  That seems like it would require a third `HashedPassword` format that represents the aggregation of the two (e.g. `|XXX|<md5hash>|<bcrypthash>`).  I admit that smells _a lot_, but it would only be required until the 3635 client is no longer supported.